### PR TITLE
Xy heatmap quantile

### PR DESCRIPTION
**Pull request by sdbernard**

### CODE REVIEW

File: xy-heatmap-quantile/index.js [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+parseData.load(dataFile, { scaleBreaks })
+    .then(({
+        seriesNames, plotData,
+    }) => { // eslint-disable-line no-unused-vars
+        Object.keys(frame).forEach((frameName) => {
+            const currentFrame = frame[frameName];
+
+            const myXAxis = gAxis.xOrdinal();// sets up yAxis
+            const myYAxis = gAxis.yOrdinal();
+            const myChart = xyHeatmapQuantileChart.draw(); // eslint-disable-line no-unused-vars
```
Comment (aendrew) [inline xy-heatmap-quantile/index.js]: Remove the eslint-disable-line comment please.

File: xy-heatmap-quantile/index.js [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+                .enter()
+                .append('g')
+                .classed('legend', true)
+                .call(myLegend);
+
+            // remove ticks from x-axis
+            myXAxis.xLabel().selectAll('.tick line').remove();
+        });
+
+    // addSVGSavers('figure.saveable');
```
Comment (aendrew) [inline xy-heatmap-quantile/index.js]: Remove extraneous comment.

File: xy-heatmap-quantile/parseData.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+function getScaleCats(el, scaleBreaks) {
+    for (let j = 0; j < scaleBreaks.length + 1; j++) {
+        if (el < scaleBreaks[j]) {
+            return j;
+        }
+
+        if (el > scaleBreaks[j] && el <= scaleBreaks[j + 1]) {
+            return j + 1;
+        }
+    }
```
Comment (aendrew) [inline xy-heatmap-quantile/parseData.js]: After the `for` loop, add `return undefined;` to fix the `consistent-return` linting error.

File: xy-heatmap-quantile/parseData.js [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+function getGroups(seriesNames, el, scaleBreaks) {
+    return seriesNames.map(name => ({
+        name,
+        value: +el[name],
+        scaleCat: getScaleCats(Number(el[name]), scaleBreaks),
+    }));
+}
+
+function getScaleCats(el, scaleBreaks) {
+    for (let j = 0; j < scaleBreaks.length + 1; j++) {
```
Comment (aendrew) [inline xy-heatmap-quantile/parseData.js]: Replace `j++` with `j += 1` to fix the `no-plusplus` linting error.

File: xy-heatmap-quantile/xyHeatmapQuantile.js [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+            .attr('height', () => yScale.bandwidth())
+            .attr('fill', d => colourScale.range()[d.scaleCat]);
+
+        if (showValues) {
+            block
+                .append('text')
+                .attr('class', 'blockValue')
+                .attr('x', d => xScale(d.name))
+                .attr('y', d => yScale(d.name))
+                .attr('dx', (xScale.bandwidth() / 2))
```
Comment (aendrew) [inline xy-heatmap-quantile/xyHeatmapQuantile.js]: The parentheses here are superfluous.

File: xy-heatmap-quantile/xyHeatmapQuantile.js [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+    };
+
+    chart.xScale = (d) => {
+        if (!d) return xScale;
+        xScale = d;
+        return chart;
+    };
+
+    chart.xDomain = (d) => {
+        xScale.domain(d);
```
Comment (aendrew) [inline xy-heatmap-quantile/xyHeatmapQuantile.js]: Should probably have a getter.

File: xy-heatmap-quantile/xyHeatmapQuantile.js [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+        return chart;
+    };
+
+    chart.xRange = (d) => {
+        xScale.rangeRound(d);
```
Comment (aendrew) [inline xy-heatmap-quantile/xyHeatmapQuantile.js]: Should probably have a getter.

File: xy-heatmap-quantile/xyHeatmapQuantile.js [hunk 8]
```diff
@@ -0,0 +1,124 @@
+import * as d3 from 'd3';
+import gChartcolour from 'g-chartcolour';
```
Comment (aendrew) [inline xy-heatmap-quantile/xyHeatmapQuantile.js]: gChartcolour is unused, please remove.

File: xy-heatmap-quantile/xyHeatmapQuantile.js (hunk 5 above)
Comment (sdbernard) [reply xy-heatmap-quantile/xyHeatmapQuantile.js]: I had them cos I was dividing the bandwidth by two. Wanted to stick to bidmas rules.

**Pull request merged**